File: src/org/baksia/rustycage/fileimport/RustImportProjectWizard.java
package org.baksia.rustycage.fileimport;

import org.eclipse.jface.viewers.IStructuredSelection;
import org.eclipse.jface.wizard.Wizard;
import org.eclipse.ui.IImportWizard;
import org.eclipse.ui.IWorkbench;

public class RustImportProjectWizard extends Wizard implements IImportWizard {
    private RustImportProjectPage rustImportProjectPage;

    public RustImportProjectWizard() {

    }

    @Override
    public void init(IWorkbench workbench, IStructuredSelection selection) {

    }

    @Override
    public boolean performFinish() {

        return false;
    }

}

File: src/org/baksia/rustycage/fileimport/RustImportProjectWizard.scala
package org.baksia.rustycage.fileimport

import org.eclipse.jface.wizard.Wizard
import org.eclipse.ui.IImportWizard
import org.eclipse.jface.viewers.IStructuredSelection
import org.eclipse.ui.IWorkbench

class RustImportProjectWizard extends Wizard with IImportWizard {
  val rustImportProjectPage: RustImportProjectPage = new RustImportProjectPage()
  setWindowTitle("Rust Project Import Wizard")
  setNeedsProgressMonitor(true)

  def init(workbench: IWorkbench, selection: IStructuredSelection) {

  }

  def performFinish(): Boolean = true

  override def addPages() {
    addPage(rustImportProjectPage)
  }
}
